### PR TITLE
Templating projects and repositories on the main page

### PR DESCRIPTION
**Pull request by tulinkry**
depends on #2718

Using jstl custom tags to implement reusable code.

### CODE REVIEW

File: opengrok-web/src/main/webapp/WEB-INF/tags/project.tag [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+fields enclosed by brackets "[]" replaced with your own identifying
+information: Portions Copyright [yyyy] [name of copyright owner]
+
+CDDL HEADER END
+
+Copyright (c) 2019, Oracle and/or its affiliates. All rights reserved.
+--%>
+<%@ tag import="org.opengrok.indexer.web.Prefix" %>
+<%@ tag import="org.opengrok.indexer.web.Util" %>
+<%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core" %>
```
Comment (vladak) [inline opengrok-web/src/main/webapp/WEB-INF/tags/project.tag]: these look like dead URLs

File: opengrok-indexer/src/main/java/org/opengrok/indexer/web/ProjectHelper.java [hunk 2]
```diff
@@ -57,6 +59,12 @@
     private static final String PROJECT_HELPER_SUBGROUPS_OF = "project_helper_subgroups_of_";
     private static final String PROJECT_HELPER_FAVOURITE_GROUP = "project_helper_favourite_group";
 
+    private static final Comparator<RepositoryInfo> REPOSITORY_COMPARATOR = new Comparator<RepositoryInfo>() {
```
Comment (vladak) [inline opengrok-indexer/src/main/java/org/opengrok/indexer/web/ProjectHelper.java]: REPOSITORY_NAME_COMPARATOR ?

File: opengrok-web/src/main/webapp/WEB-INF/tags/project.tag [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+When distributing Covered Code, include this CDDL HEADER in each
+file and include the License file at LICENSE.txt.
+If applicable, add the following below this CDDL HEADER, with the
+fields enclosed by brackets "[]" replaced with your own identifying
+information: Portions Copyright [yyyy] [name of copyright owner]
+
+CDDL HEADER END
+
+Copyright (c) 2019, Oracle and/or its affiliates. All rights reserved.
```
Comment (vladak) [inline opengrok-web/src/main/webapp/WEB-INF/tags/project.tag]: add your copyright there (for all such files)

File: opengrok-web/src/main/webapp/WEB-INF/tags/project.tag (hunk 1 above)
Comment (tulinkry) [reply opengrok-web/src/main/webapp/WEB-INF/tags/project.tag]: the joke here is that this does not point to an external URI but identifies the tag library on the class path:

from jstl-1.2.jar

```
  <description>JSTL 1.1 core library</description>
  <display-name>JSTL core</display-name>
  <tlib-version>1.1</tlib-version>
  <short-name>c</short-name>
  <uri>http://java.sun.com/jsp/jstl/core</uri>
```

File: opengrok-indexer/src/main/java/org/opengrok/indexer/web/ProjectHelper.java (hunk 2 above)
Comment (tulinkry) [reply opengrok-indexer/src/main/java/org/opengrok/indexer/web/ProjectHelper.java]: sure

File: opengrok-web/src/main/webapp/WEB-INF/tags/project.tag (hunk 3 above)
Comment (tulinkry) [reply opengrok-web/src/main/webapp/WEB-INF/tags/project.tag]: done

File: opengrok-web/src/main/webapp/WEB-INF/tags/project.tag [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+Portions Copyright (c) 2019, Krystof Tulinger <k.tulinger@seznam.cz>.
+--%>
+<%@ tag import="org.opengrok.indexer.web.Prefix" %>
+<%@ tag import="org.opengrok.indexer.web.Util" %>
+<%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core" %>
+<%@ taglib uri="http://java.sun.com/jsp/jstl/functions" prefix="fn" %>
+<%@ taglib prefix="opengrok" tagdir="/WEB-INF/tags" %>
+
+<%@ attribute name="pageConfig" required="true" type="org.opengrok.indexer.web.PageConfig" %>
```
[the remark this answers is not in the document]
Comment (tulinkry) [reply opengrok-web/src/main/webapp/WEB-INF/tags/project.tag]: will remove

**Comment by vladak** [comment]
Nice work !

**Pull request merged**